graylog2-web-interface/src/views/components/contexts/WidgetFocusProvider.tsx

```diff
@@ -70,58 +70,81 @@ const _updateQueryParams = (
   return baseUri.toString();
 };
 
-const useSyncStateWithQueryParams = ({ focusedWidget, focusUriParams, setFocusedWidget, widgets }) => {
+type FocusUriParams = {
+  editing: boolean,
+  focusing: boolean,
+  id: string,
+  isPageShown: boolean,
+};
+type SyncStateArgs = {
+  focusedWidget: FocusContextState | undefined,
+  setFocusedWidget: (newFocusedWidget: FocusContextState | undefined) => void,
+  widgetIds: Array<string>,
+  focusUriParams: FocusUriParams,
+}
+
+const useSyncStateWithQueryParams = ({ focusedWidget, focusUriParams, setFocusedWidget, widgetIds }: SyncStateArgs) => {
+  const dispatch = useAppDispatch();
+
   useEffect(() => {
     const nextFocusedWidget = {
       id: focusUriParams.id,
       editing: focusUriParams.editing,
       focusing: focusUriParams.focusing || focusUriParams.editing,
-    };
+    } as FocusContextState;
 
     if (!isEqual(focusedWidget, nextFocusedWidget)) {
-      if (focusUriParams.id && !widgets.has(focusUriParams.id)) {
+      if (focusUriParams.id && !widgetIds.includes(focusUriParams.id)) {
         return;
       }
 
       setFocusedWidget(nextFocusedWidget);
       const filter = nextFocusedWidget?.id ? [nextFocusedWidget.id] : null;
```

Comment:
Let's change `null` to `undefined`, to avoid a state update?
<img width="596" alt="image" src="https://user-images.githubusercontent.com/46300478/218725331-3d2b47ba-8415-47bc-9600-b8ccbcfba944.png">